### PR TITLE
Add DUO importer

### PR DESCRIPTION
**Pull request by anibyl**
Hey there!

This PR adds [DUO Mobile](https://play.google.com/store/apps/details?id=com.duosecurity.duomobile) OTP importer.

PS: I added translations in the languages I knew; please let me know if it's not the right way of contributing translations.

#68

### CODE REVIEW

**Review by alexbakker — CHANGES_REQUESTED** [review]
Thanks for working on this! It'll take me a couple of days before I can get my hands on a rooted device again, so I can't test this immediately, but here are some initial comments.

File: app/src/main/java/com/beemdevelopment/aegis/importers/DuoImporter.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            _array = array;
+        }
+
+        @Override
+        public @NonNull Result convert() throws DatabaseImporterException {
+            @NonNull Result result = new Result();
+
+            try {
+                for (int i = 0; i < _array.length(); i++) {
+                    @NonNull JSONObject entry = _array.getJSONObject(i);
```
[the remark this answers is not in the document]
Comment (anibyl) [reply app/src/main/java/com/beemdevelopment/aegis/importers/DuoImporter.java:67]: This is something we use at work. Removed.

File: app/src/main/java/com/beemdevelopment/aegis/importers/DuoImporter.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            try {
+                @NonNull String label = entry.optString("name");
+
+                @NonNull JSONObject otpData = entry.getJSONObject("otpGenerator");
+
+                @NonNull byte[] secret = Base32.decode(otpData.getString("otpSecret"));
+
+                @Nullable Long counter;
+                try {
+                    counter = otpData.getLong("counter");
```
[the remark this answers is not in the document]
Comment (anibyl) [reply app/src/main/java/com/beemdevelopment/aegis/importers/DuoImporter.java:93]: Mostly my unfamiliarity with org.json library.
I wanted to use `#optLong(String, long)` but it receives a primitive as a fallback which is unacceptable: I need a nullable variable.
Replaced with `#has` and `#getLong`.

**Comment by anibyl** [comment]
Hi! How's your search for a rooted device going?

**Comment by alexbakker** [comment]
Yes, works well! Could you squash everything into a single commit? I'll merge this afterwards.

**Comment by Hacker437** [comment]
> Yes, works well! Could you squash everything into a single commit? I'll merge this afterwards.

@alexbakker I think you can squash when merging 🤔?

**Comment by alexbakker** [comment]
Yes, but "squash and merge" performs a fast-forward merge instead of creating a merge commit. I'd like to keep things consistent.

**Comment by Hacker437** [comment]
> Yes, but "squash and merge" performs a fast-forward merge instead of creating a merge commit. I'd like to keep things consistent.

Ohho ok.

**Comment by anibyl** [comment]
> Yes, works well! Could you squash everything into a single commit? I'll merge this afterwards.

Done.

**Review by alexbakker — APPROVED** [review]
Thanks!

**Pull request merged**